### PR TITLE
3.0.0

### PR DESCRIPTION
**Pull request by pkesavap**
@gobindadas @sabose 

The problem which caused  issue #544 , was because of a silent exception raised when an entity which was not there in the dictionary was removed ,that resulted in failure of invoking further methods in gdeploy leading to cause the issue #544 .

This change checks if the entity is there in the dictionary before removing it.

### CODE REVIEW

**Comment by sabose** [comment]
@sac could you review as well?

File: gdeploycore/backend_setup.py [hunk 1]
```diff
@@ -141,6 +141,8 @@ def parse_section(self, hostname):
         try:
             self.section_dict = Global.sections['backend-setup' +
                     hostname]
+            if '__name__' in self.section_dict.keys():
+                del self.section_dict['__name__']
```
Comment (sabose) [inline gdeploycore/backend_setup.py:145]: is __name__ added to dict.keys anywhere?

Comment (sac) [reply gdeploycore/backend_setup.py:145]: This change is not needed. I happened to look into the issue 544 now. The conf file is written is wrong. If you have to use multiple vgs they have to be comma-separated. 
I'll comment on the issue.

**Review by sac — CHANGES_REQUESTED** [review]
This change is not needed. The conf file is written wrongly.

**Comment by pkesavap** [comment]
> 

> This change is not needed. The conf file is written wrongly.

Hey sac, 
With  the following conf file(with proper conf file) , the error is still seen :- 
[Hosts]
host.hostname.com
[backend-setup] 
devices=/dev/vdb,/dev/vdc,/dev/vdd,...vdm
 vgs=vg0,vg1,vg2,vg3,vg4,vg5,vg6,vg7,vg8,vg9,vg10,vg11
pools=pool0,pool1,pool2,pool3,pool4,pool5,pool6,pool7,pool8,pool9,pool10,pool11
lvs=lv0,lv1,lv2,lv3,lv4,lv5,lv6,lv7,lv8,lv9,lv10,lv11
brick_dirs=/bricks/brick0,/bricks/brick1,/bricks/brick2,/bricks/brick3,/bricks/brick4,/bricks/brick5,/bricks/brick6,/bricks/brick7,/bricks/brick8,/bricks/brick9,/bricks/brick10,/bricks/brick11

__________________
the following error(snapshot) is seen:-
TASK [Create Physical Volume] ********************************************************
failed: [dhcp37-52.lab.eng.blr.redhat.com] (item=/dev/vdb,/dev/vdc,/dev/vdd,/dev/vde,/dev/vdf,/dev/vdg,/dev/vdh,/dev/vdi,/dev/vdj,/dev/vdk,/dev/vdl,/dev/vdm) => {"ansible_facts": {"discovered_interpreter_python": "/usr/libexec/platform-python"}, "ansible_loop_var": "item", "changed": false, "failed_when_result": true, "item": "/dev/vdb,/dev/vdc,/dev/vdd,/dev/vde,/dev/vdf,/dev/vdg,/dev/vdh,/dev/vdi,/dev/vdj,/dev/vdk,/dev/vdl,/dev/vdm", "msg": "  Device /dev/vdb,/dev/vdc,/dev/vdd,/dev/vde,/dev/vdf,/dev/vdg,/dev/vdh,/dev/vdi,/dev/vdj,/dev/vdk,/dev/vdl,/dev/vdm not found.\n", "rc": 5}
failed: [dhcp37-187.lab.eng.blr.redhat.com] (item=/dev/vdb,/dev/vdc,/dev/vdd,/dev/vde,/dev/vdf,/dev/vdg,/dev/vdh,/dev/vdi,/dev/vdj,/dev/vdk,/dev/vdl,/dev/vdm) => {"ansible_facts": {"discovered_interpreter_python": "/usr/libexec/platform-python"}, "ansible_loop_var": "item", "changed": false, "failed_when_result": true, "item": "/dev/vdb,/dev/vdc,/dev/vdd,/dev/vde,/dev/vdf,/dev/vdg,/dev/vdh,/dev/vdi,/dev/vdj,/dev/vdk,/dev/vdl,/dev/vdm", "msg": "  Device /dev/vdb,/dev/vdc,/dev/vdd,/dev/vde,/dev/vdf,/dev/vdg,/dev/vdh,/dev/vdi,/dev/vdj,/dev/vdk,/dev/vdl,/dev/vdm not found.\n", "rc": 5}

**Review by gobindadas — APPROVED** [review]

**Comment by gobindadas** [comment]
As scratch build is working as expected, so merging this.

**Pull request merged**